src/module.c

```diff
@@ -1355,6 +1375,29 @@ int RM_BlockedClientMeasureTimeEnd(RedisModuleBlockedClient *bc) {
     return REDISMODULE_OK;
 }
 
+/* This API allows modules to let Redis process some commands during long
+ * blocking execution of a module command. The module can call this API
+ * periodically, and after the time defined by the `script-time-limit` config,
+ * Redis will start rejecting most commands with `-BUSY` error, but allow the
+ * ones marked with the `allow-busy` flag to be executed. This API can also be
+ * used in thread safe context (while locked), and during loading (in the
+ * rdb_load, in which case it'll reject commands with -LOADING error) */
+void RM_Yield(RedisModuleCtx *ctx, const char *busy_reply) {
+    long long now = getMonotonicUs();
+    if (now >= ctx->next_yield_time) {
+        const char *prev_busy_module_yield_reply = server.busy_module_yield_reply;
+        server.busy_module_yield_reply = busy_reply;
```

Comment:
I don't like the fact that passing a null `busy_reply` will *sometimes* assert (only if we hit the code path to actually produce that reply).

I suggest to either use a default, assert here instead, or maybe just have `RM_Yield` return an error.